### PR TITLE
asm-lsp: add support for darwin

### PR DESCRIPTION
**Pull request by caiquefigueiredo**
## Description of changes

Support darwin for asm-lsp

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [X] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by NotAShelf** [comment]
Please apply formatting in a different commit. Also, you might want to add yourself as a maintainer if Darwin support is to be feasible: I do not own any Darwin hardware, and I cannot
guarantee support for non-Linux.

File: pkgs/by-name/as/asm-lsp/package.nix [hunk 1]
```diff
@@ -18,13 +22,16 @@ rustPlatform.buildRustPackage {
...
   };
 
-  nativeBuildInputs = [
-    pkg-config
-  ];
+  nativeBuildInputs =
+    [
+      pkg-config
+    ]
+    ++ lib.optionals stdenv.buildPlatform.isDarwin [
```
Comment (FrothyMarrow) [inline pkgs/by-name/as/asm-lsp/package.nix:29]: Dependencies such as libraries/frameworks need to be added to `buildInputs`. You can find more information [here](https://nixos.org/manual/nixpkgs/stable/#ssec-stdenv-dependencies-reference).

**Review by FrothyMarrow — CHANGES_REQUESTED** [review]

**Comment by caiquefigueiredo** [comment]
> Please apply formatting in a different commit.

@NotAShelf could you elaborate on this, please? Do you wish me to apply a specific formatting for the changes in an additional commit? By "formatting" you mean a standard way of linting nix files or something else?

> you might want to add yourself as a maintainer if Darwin

Will do. Just gonna wait on the previous point so that I can put it all in a single commit

**Comment by khaneliman** [comment]
Result of `nixpkgs-review pr 344556` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>asm-lsp</li>
  </ul>
</details>

**Comment by khaneliman** [comment]
> Also, you might want to add yourself as a maintainer if Darwin support is to be feasible: I do not own any Darwin hardware, and I cannot guarantee support for non-Linux.

Can always just ping the darwin maintainers if you ever get tagged with an issue with it working for darwin, too.

**Comment by NotAShelf** [comment]
> > Please apply formatting in a different commit.
> 
> @NotAShelf could you elaborate on this, please? Do you wish me to apply a specific formatting for the changes in an additional commit? By "formatting" you mean a standard way of linting nix files or something else?

You are changing the formatting of the file in your commit, from nixpkgs-fmt to nixfmt. I'd prefer if formatting changes and platform changes were done in two separate commits.

**Comment by caiquefigueiredo** [comment]
Sorry for the back and forth, @NotAShelf. Hopefully everything is ok now 🙏🏼 
Went ahead and added me as a maintainer. Please let me know if something else seems off

**Comment by caiquefigueiredo** [comment]
@NotAShelf ping

**Review by NotAShelf — APPROVED** [review]

**Comment by NotAShelf** [comment]
diff LGTM, @FrothyMarrow for testing

**Review by FrothyMarrow — APPROVED** [review]

**Comment by caiquefigueiredo** [comment]
Merge ok?

**Pull request merged**